internal/webhook/managedcluster_webhook.go

```diff
@@ -84,6 +97,15 @@ func (v *ManagedClusterValidator) ValidateUpdate(ctx context.Context, _ runtime.
 	if err != nil {
 		return nil, fmt.Errorf("%s: %v", errInvalidManagedCluster, err)
 	}
+	if !newManagedCluster.Spec.DryRun && templateutil.IsManagedByHMC(template) {
```

Comment:
We wanted to have a list of available upgrades in the managedcluster status and only check that the requested template is in the list in admission